### PR TITLE
fix undefined SapphireTest::using_temp_db()

### PR DESCRIPTION
**Pull request by wernerkrauss**
see #33

### CODE REVIEW

**Comment by dhensby** [comment]
@wernerkrauss we're hoping we can reduce coupling of non-test related code from referencing the test classes/state. Is there a way we can solve this without having to look at the test state?

In core the MySQL class doesn't seem to care one way or another https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/Connect/MySQLSchemaManager.php#L147-L171

File: code/SQLite3SchemaManager.php [hunk 1]
```diff
@@ -210,7 +210,7 @@ public function checkAndRepairTable($tableName = null)
     {
         $ok = true;
 
-        if (!SapphireTest::using_temp_db() && !self::$checked_and_repaired) {
+        if (SapphireTest::getTempDB()->isUsed() && !self::$checked_and_repaired) {
```
Comment (dhensby) [inline code/SQLite3SchemaManager.php]: You've missed out the `!` from the first condition

File: code/SQLite3SchemaManager.php [hunk 2]
```diff
@@ -210,7 +210,7 @@ public function checkAndRepairTable($tableName = null)
     {
         $ok = true;
 
-        if (!SapphireTest::using_temp_db() && !self::$checked_and_repaired) {
+        if (!SapphireTest::getTempDB()->isUsed() && !self::$checked_and_repaired) {
```
Comment (tractorcow) [inline code/SQLite3SchemaManager.php]: We need to remove SapphireTest references from non-test code. This entire method will fail if phpunit isn't installed, which is often the case in production.

**Comment by tractorcow** [comment]
>@wernerkrauss we're hoping we can reduce coupling of non-test related code from referencing the test classes/state. Is there a way we can solve this without having to look at the test state?

I'm raising the priority of this from "hope" to "need to".

File: code/SQLite3SchemaManager.php (hunk 2 above)
Comment (tractorcow) [reply code/SQLite3SchemaManager.php]: You can just use `TempDatabase::singleton()->isUsed()` if necessary here.

Comment (wernerkrauss) [reply code/SQLite3SchemaManager.php]: Thanks. Does #35 make this PR unnecessary?

Comment (tractorcow) [reply code/SQLite3SchemaManager.php]: Yes I think so. :)

**Pull request closed without merging**